### ISSUE TITLE
tv_grab_eu_xmltvse is broken, possible dirty fix using EPG xmltv files from iptv.org

### ISSUE THREAD

**Issue by Supermagnum** [XMLTV/xmltv#161]
Hello, since the XML grabber in tvheadend from the truecharts catalogue is currently broken , I think that it is possible to download the required XML files and import those with /usr/bin./tv_grab_file .
The bug is reported here: 
https://github.com/XMLTV/xmltv/issues/152

I have tried to write a script that should be possible to run with a daily  crontab entry,- but since there seems not to be any possibility to edit the necessary file I think that it can be downloaded to the internal pod with wget. 

The script is here: https://github.com/Supermagnum/xmlimport/blob/main/tvgrabber.txt

It uses EPG information from:
https://github.com/iptv-org/epg

I hope that it makes sense, and I haven't made any errors. 

So, any comments?

**Comment by garybuhrmaster** [comment]
> So, any comments?

Historically, iptv-org has had a (very very) lax attitude regarding of authorization/approval of their sources (they like to assert they do not validate anything, but trust any submissions).  The last time I was aware of of an attempt to validate their sources over half were clearly not legal (piracy, etc.), a number were of unknown authorization, and only a small percentage were clearly legal (almost so small as to be almost noise in the total number(s) of sources in some locations).

The project itself has historically required clear TOUs that allow access to the data.  Can you provide an authoritative statement from all those EPG sources (themselves, not iptv-org) that they have approved access to that EPG that iptv-org is making available?

Individuals, of course, may individually choose to respect patent/copyright/trademark/TOU, and they may choose not to, and any jurisdictional requirements will apply to those individuals. for their choices.

**Comment by Supermagnum** [comment]
> > So, any comments?
> 
> Historically, iptv-org has had a (very very) lax attitude regarding of authorization/approval of their sources (they like to assert they do not validate anything, but trust any submissions). The last time I was aware of of an attempt to validate their sources over half were clearly not legal (piracy, etc.), a number were of unknown authorization, and only a small percentage were clearly legal (almost so small as to be almost noise in the total number(s) of sources in some locations).
> 
> The project itself has historically required clear TOUs that allow access to the data. Can you provide an authoritative statement from all those EPG sources (themselves, not iptv-org) that they have approved access to that EPG that iptv-org is making available?
> 
> Individuals, of course, may individually choose to respect patent/copyright/trademark/TOU, and they may choose not to, and any jurisdictional requirements will apply to those individuals. for their choices.

No, but it seems like it's grabbed from official website's according to:
https://github.com/iptv-org/epg/tree/master/sites

For example, this looks like public information:
https://www.allente.se/tv-guide/

**Comment by honir** [comment]
> For example, this looks like public information:

Just because it is on the internet does not make something "public".

It's not that the information is visible on the website but how they authorise you to **use** it. That is copyright. 

You would need to find **Allente**'s Terms and Conditions to see how you may use the data (i.e. whether you can download (copy) it or not, and what you may do with it).

**Comment by rmeden** [comment]
Schedules Direct was recently asked to generate custom lineups based on 
the IPTV.org github and we ended up declining... too many sketchy 
sources and we don't want to make things easier for pirates. We know SD 
is sometimes used by pirates (and data from us is pirated too!), but 
there's little we can do... but we certainly don't want to encourage it.

**Comment by Supermagnum** [comment]
The only thing I can find is about their equipment, and services. There is nothing that mentions the EPG in those.

**Comment by garybuhrmaster** [comment]
> The only thing I can find is about their equipment, and services. There is nothing that mentions the EPG in those.

In which case the Berne convention applies (at least in all countries that are signatories, which include this source), which means all content is "born" copyrighted (and not automatically free to use without license).

I would recommend you contact the origin of the content (corporate offices, possibly corporate legal) and request that they clarify their TOU/TOS should they wish to donate their content to the public.  Some organizations will do so, others, not so much.  Their (copyrighted) content, their choices.
